program-runtime/src/timings.rs

```diff
@@ -35,16 +35,112 @@ impl ProgramTiming {
     }
 }
 
+#[derive(Default, Debug)]
+pub struct ExecuteTimings {
+    pub check_us: u64,
+    pub load_us: u64,
+    pub execute_us: u64,
+    pub store_us: u64,
+    pub update_stakes_cache_us: u64,
+    pub total_batches_len: usize,
+    pub num_execute_batches: u64,
+    pub collect_logs_us: u64,
+    pub details: ExecuteDetailsTimings,
+    pub execute_accessories: ExecuteAccessoryTimings,
+}
+
+impl ExecuteTimings {
```

Comment:
Another way to do this would be indexed arrays where the index is a known type, that might simplify the timings API and be more clearly expandable.

Reply:
Could be an `enum` and then do `Enum::MyMetric as usize`

Reply:
That would turn 
```
invoke_context.timings.create_executor_register_syscalls_us = invoke_context
        .timings
        .create_executor_register_syscalls_us
        .saturating_add(register_syscalls_time.as_us());
        ```
into:
```
invoke_context.timings.accumulate(CREATE_EXECUTOR_REGISTER_SYSCALLS_US, register_syscalls_time.as_us());
```

Reply:
Agreed that these timings are getting a bit out of control, but this set just moves `ExecuteTimings` from `runtime::bank`  to `program_runtime::timings`.  I'll add an issue suggesting future refactor

Reply:
Done in. https://github.com/solana-labs/solana/issues/22325